src/UserRepository.js

```diff
@@ -4,9 +4,25 @@ class UserRepository {
   }
 
   getUser(id) {
-    return this.users.find(function (user) {
-      return user.id === id;
+    return this.users.find((user) => id === user.id) 
+  }
+
+  calculateAverageStepGoal() {
+    let goals = this.users.map((user) => {
+      return user.dailyStepGoal;
     });
+    let total = goals.reduce((sum, goal) => {
+      sum += goal;
+      return sum;
+    }, 0);
+    return total / this.users.length;
+  }
+
+  findFriendsNames(userFriends) {
+    let friendsNames = [];
+    userFriends.forEach((id) => {
+      this.getUser(id)
+    })
```

Comment:
We can move this method to user and have user inherit the user repo class if we want. other wise we have this function in the update dom file right now